cmd/hook.go

```diff
@@ -669,7 +671,8 @@ Gitea or set your environment appropriately.`, "")
 		if err != nil {
 			return err
 		}
-		if rs.OldOID != git.EmptySHA {
+		commitID, _ := git.IDFromString(rs.OldOID)
+		if !commitID.IsZero() {
```

Comment:
I meant, could it be `if !git.IsEmptyOID(rs.OldOID)` ? Only 1 line, just like before

Reply:
I would rather not since that would imply CommitID == string and ideally it would be CommitID after it got parsed at all times. If Gitea was refactored further where all CommitIDs would no longer be (potentially random) strings, I believe the codebase would be a lot more consistent.

But I can add this helper function if this is a real sticking point here. Let me know.

Reply:
I did this change but I've only found 2 places where this is useful. The other places seem to benefit from distinction between different hash lengths. But not certain  if my understanding is correct. Open to further suggestions.